project/frontend/src/components/RestaurantCardStack.js

```diff
@@ -0,0 +1,89 @@
+import './RestaurantCardStack.css';
+
+import React, { useState } from 'react';
+
+import RestaurantCard from './RestaurantCard.js';
+
+/**
+ * A navigable stack of restaurant cards.
+ *
+ * @param {!Array<Object<string, *>>} props.cards A list of restaurant objects whose
+ * data will be displayed in RestaurantCard components within this stack.
+ */
+function RestaurantCardStack(props) {
+  const [currentCardIndex, setCurrentCardIndex] = useState(0);
+
+  const maxScaleFactor = 1.0;
+  const minScaleFactor = 0.85;
+  const numberOfCollapsedCards = 3;
+  const uncollapsedCardHeight = 487;
+  const offsetBetweenCollapsedCards = 24;
+  const offsetBeforeUncollapsedCard = 152;
+  const transitionString =
+    'top 0.75s cubic-bezier(0.35, 0.91, 0.33, 0.97), ' +
+    'opacity 0.75s cubic-bezier(0.35, 0.91, 0.33, 0.97), ' +
+    'transform 0.75s cubic-bezier(0.35, 0.91, 0.33, 0.97)';
+  const cards = props.cards.map((card, index) => {
+    if (index < currentCardIndex - 7 || index > currentCardIndex + 7) {
+      return null;
+    }
+
+    const relativeIndex = index - currentCardIndex;
+    const relativeIndexAbs = Math.abs(relativeIndex);
+    const offset =
+      relativeIndex === 0
+        ? offsetBeforeUncollapsedCard
+        : relativeIndex > 0
+        ? offsetBeforeUncollapsedCard +
+          uncollapsedCardHeight +
+          Math.min(relativeIndex, numberOfCollapsedCards) *
+            offsetBetweenCollapsedCards
+        : Math.max(numberOfCollapsedCards + relativeIndex, 0) *
+          offsetBetweenCollapsedCards;
```

Comment:
errr this is a bit of personal pref but I think chained ternaries especially given that the expressions within a clause aren't trivial here -- are a bit hard to follow while reading -- consider just using conditionals instead? 
```
const offset;
if ( relativeIndex === 0):
    offset = offsetBeforeUncollapsedCard;
else if (relativeIndex > 0):
    offset = offsetBeforeUncollapsedCard + uncollapsedCardHeight +
          Math.min(relativeIndex, numberOfCollapsedCards) * offsetBetweenCollapsedCards;
else:
   offset =  Math.max(numberOfCollapsedCards + relativeIndex, 0) * offsetBetweenCollapsedCards;
```